components/org.wso2.carbon.identity.tag.manager.core/src/main/java/org/wso2/carbon/identity/tag/manager/core/dao/TagManagementDAOImpl.java

```diff
@@ -0,0 +1,361 @@
+        }
+    }
+
+    @Override
+    public TagAssociationsResult loadTagAssociationsByResourceId(String resourceUuid)
+            throws TagServiceException {
+
+        List<Tag> tagList = new ArrayList<>();
+        PreparedStatement prepStmt;
+        ResultSet rs;
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
+            prepStmt = connection.prepareStatement(Constants.SQLConstants.GET_TAG_RESOURCE_ASC_BY_RESOURCE_UUID_SQL);
+            prepStmt.setString(1, resourceUuid);
+            rs = prepStmt.executeQuery();
+            while (rs.next()) {
+                Tag tag = new Tag();
+                tag.setTagUuid(rs.getString(1));
+                tag.setName(rs.getString(2));
+                tag.setPubliclyVisible(isPublic(rs.getString(3)));
+                tagList.add(tag);
+            }
+        } catch (SQLException e) {
+            throw new TagServiceException(ErrorMessage.ERROR_CODE_ERROR_GETTING_ASSOCIATIONS.getCode(), e);
+        }
+        return new TagAssociationsResult(tagList.size(), tagList);
+    }
+
+    @Override
+    public void deleteAssociationByResource(String resourceUuid) throws TagServiceException {
+
+        PreparedStatement prepStmt;
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
+            prepStmt = connection.prepareStatement(Constants.SQLConstants.DELETE_TAG_RESOURCE_ASC_BY_RESOURCE_SQL);
+            prepStmt.setString(1, resourceUuid);
+            prepStmt.execute();
+            IdentityDatabaseUtil.commitTransaction(connection);
+        } catch (SQLException e) {
+            throw new TagServiceException(ErrorMessage.ERROR_CODE_ERROR_DELETING_ASSOCIATION.getCode(),
+                    ErrorMessage.ERROR_CODE_ERROR_DELETING_ASSOCIATION.getMessage(), e);
+        }
+    }
+
+    @Override
+    public boolean isExistingAssociation(String tagUuid, String resourceId) throws TagServiceException {
+
+        PreparedStatement prepStmt;
+        boolean status;
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            prepStmt = connection.prepareStatement(Constants.SQLConstants.GET_TAG_RESOURCE_ASC_BY_RESOURCE_TAG_UUID_SQL);
+            prepStmt.setString(1, tagUuid);
+            prepStmt.setString(1, resourceId);
+            status = prepStmt.execute();
+        } catch (SQLException e) {
+            throw new TagServiceException(ErrorMessage.ERROR_CODE_ERROR_GETTING_ASSOCIATION.getCode(), e);
+        }
+        return status;
+    }
+
+    @Override
+    public boolean isExistingTagId(String tagUuid) throws TagServiceException {
+
+        PreparedStatement prepStmt;
+        boolean status;
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            prepStmt = connection.prepareStatement(Constants.SQLConstants.GET_IDN_TAG_BY_TAG_UUID_SQL);
+            prepStmt.setString(1, tagUuid);
+            status = prepStmt.execute();
```

Comment:
Can we just check the existence without actually loading all the attributes?

Reply:
fixed